pkg/oc/admin/policy/modify_roles.go

```diff
@@ -531,20 +531,26 @@ func (o *RoleModificationOptions) RemoveRole() error {
 	subjectsToRemove := authorizationapi.BuildSubjects(o.Users, o.Groups)
 	subjectsToRemove = append(subjectsToRemove, o.Subjects...)
 
-	if len(o.Output) > 0 {
+	found := false
+	if len(o.Output) > 0 || o.DryRun {
 		for _, binding := range roleBindings {
-			binding.Subjects = removeSubjects(binding.Subjects, subjectsToRemove)
+			binding.Subjects, found = removeSubjects(binding.Subjects, subjectsToRemove)
+			if !found {
+				return fmt.Errorf("unable to find target %v", o.Targets)
+			}
 			updatedBindings.Items = append(updatedBindings.Items, *binding)
 		}
+		if o.DryRun {
```

Comment:
This changes existing behavior - it makes DryRun not PrintObj which is incorrect.